File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int ans = Integer.MIN_VALUE;
    public int maxPathSum(TreeNode root) {
        
        dfs(root);
        return ans;

        
    }

    private int dfs(TreeNode n){
        if(n == null)return 0;
        int left = Math.max(0, dfs(n.left));
        int right = Math.max(0, dfs(n.right));

        ans = Math.max(ans, n.val + left + right);

        return n.val + Math.max(right, left);
    }
}

File: Data Structures & Algorithms/delete-node-in-a-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minNode(self, root:Optional[TreeNode]):
        cur = root
        while cur and  cur.left:
            cur = cur.left
        return cur
    def deleteNode(self, root: Optional[TreeNode], key: int) -> Optional[TreeNode]:
        if not root: return None

        if root.val< key : 
            root.right = self.deleteNode(root.right, key)
        elif root.val> key : 
            root.left = self.deleteNode(root.left, key)

        else: 
            if not root.right: return root.left
            elif not root.left: return root.right
            else: 
                minval = self.minNode(root.right)
                succ = minval.val
                root.val = succ
                root.right = self.deleteNode(root.right, succ )

        return root


        